detox/android/detox/src/main/java/com/wix/detox/common/UIThread.kt

```diff
@@ -0,0 +1,69 @@
+package com.wix.detox.common
+
+import android.os.Handler
+import android.os.Looper
+import java.util.concurrent.Callable
+import java.util.concurrent.FutureTask
+
+object UIThread {
```

Comment:
Could you elaborate about this class? I am curious.

Reply:
Sure! The main thread is a first-class citizen on Android, and there is special interest in enqueuing tasks to run over it (in various flavours - queue back, queue front, sync/async...). There are many ways to achieve access to the main thread, but they're not all accessible in all contexts. So, the easiest solution is to unify this functionality onto one object that enables to do that anywhere.